tests/test_task.py

```diff
@@ -1046,3 +1046,114 @@ def test_task_beatpos_tail(SR, HOP_LENGTH, SPARSE):
                            axis=0).astype(Y_pred.dtype)
     for i, (y1, y2) in enumerate(zip(Y_pred, Y_expected)):
         assert y1 == y2
+
+
+def test_task_key__encode_key_str(SPARSE):
+    # Checks the helper function which does key string to encoding
+    trans = pumpp.task.KeyTransformer(sparse=SPARSE)
+    
+    # Check A:minor
+    pitch_profile, tonic = trans._encode_key_str('A:minor')
+    assert np.all(pitch_profile == np.array([1, 0, 1, 0, 1, 1, 0, 1, 0, 1, 0, 1]))
+    if SPARSE:
+        assert tonic == 9
+    else:                            #   C #C  D bE  E  F #F  G bA  A bB  B  N
+        assert np.all(tonic == np.array([0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0]))
+
+    # Check 'N' for no key
+    pitch_profile, tonic = trans._encode_key_str('N')
+    assert np.all(pitch_profile == np.array(np.zeros(12, dtype=np.bool)))
+    # Or should the above the all 1's with np.ones()?
+    if SPARSE:
+        assert tonic == 12
+    else:
+        assert np.all(tonic == np.array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1]))
+ 
+        
+def test_task_key_fields(SPARSE):
+    trans = pumpp.task.KeyTransformer(name='mykey', sparse=SPARSE)
+
+    assert set(trans.fields.keys()) == set(['mykey/pitch_profile',
+                                            'mykey/tonic'])
+
+    assert trans.fields['mykey/pitch_profile'].shape == (None, 12)
+    assert trans.fields['mykey/pitch_profile'].dtype is np.bool
+
+    if SPARSE:
+        assert trans.fields['mykey/tonic'].shape == (None, 1)
+        assert np.issubdtype(trans.fields['mykey/tonic'].dtype, np.integer)
+    else:
+        assert trans.fields['mykey/tonic'].shape == (None, 13)
+        assert trans.fields['mykey/tonic'].dtype is np.bool
+
+
+def test_task_key_present(SR, HOP_LENGTH, SPARSE):
+    # Create jams with key annotation
+    jam = jams.JAMS(file_metadata=dict(duration=12.0))
+
+    ann = jams.Annotation(namespace='key_mode')
+
+    ann.append(time=0, duration=2.0, value='C:minor')
+    ann.append(time=2, duration=2.0, value='N')
+    ann.append(time=4, duration=2.0, value='Eb')
+    ann.append(time=8, duration=2.0, value='D:major')
+    ann.append(time=10, duration=2.0, value='D:lydian')
+
+    jam.annotations.append(ann)
+
+    trans = pumpp.task.KeyTransformer(name='key',
+                                      sr=SR, hop_length=HOP_LENGTH,
+                                      sparse=SPARSE)
+    
+    output = trans.transform(jam)
+
+    # Make sure we have the mask
+    assert np.all(output['key/_valid'] == [0, 12 * trans.sr // trans.hop_length])
+
+    # Ideal vectors:
+    # pcp = Cmin, N, Eb, N, D, D_lyd TODO
+    pcp_true = np.array([
+        trans._encode_key_str('C:minor')[0],
+        trans._encode_key_str('N')[0],
+        trans._encode_key_str('Eb:major')[0],
+        trans._encode_key_str('N')[0],
+        trans._encode_key_str('D:major')[0],
+        trans._encode_key_str('D:lydian')[0]
```

Comment:
can you add an example that tests an implicit major key like `'G#'`?

Reply:
Yes